### PR TITLE
OBJ-249 check application user details

### PR DESCRIPTION
**Pull request by mattch1**
User information required on check answers screen.

email address
full name
share/divulge information

### CODE REVIEW

File: views/check-your-answers.html [hunk 1]
```diff
@@ -28,6 +28,57 @@ <h1 class="govuk-heading-xl">
...
+            key: {
+              text: "Full name"
+            },
+            value: {
+              text: objection.created_by.fullName
+            },
+            actions: {
+              items: [
+                {
+                  href: "/objecting-entity-name/?id=1",
```
Comment (mwestacott) [inline views/check-your-answers.html:55]: Looks like this is adding the 'change' link but not in screenshot on story or part of description.

File: views/check-your-answers.html [hunk 2]
```diff
@@ -28,6 +28,57 @@ <h1 class="govuk-heading-xl">
...
+            key: {
+              text: "Share identity if company requests it?"
+            },
+            value: {
+              text: objection.created_by.shareIdentity
+            },
+            actions: {
+              items: [
+                {
+                  href: "/objecting-entity-name/?id=1",
```
Comment (mwestacott) [inline views/check-your-answers.html:72]: Again, I think 'change' functionality is covered by another story?

**Comment by ICoxCH** [comment]
It seems your latest commit has caused a couple unit tests to fail

**Comment by markpit** [comment]
unit tests are failing

File: src/controllers/check.your.answers.controller.ts [hunk 3]
```diff
@@ -2,7 +2,7 @@ import { Session } from "ch-node-session-handler";
 import { NextFunction, Request, Response } from "express";
 import { OBJECTIONS_CONFIRMATION } from "../model/page.urls";
 import { Templates } from "../model/template.paths";
-import { Objection } from "../modules/sdk/objections";
+import {CreatedBy, Objection} from "../modules/sdk/objections";
```
Comment (markpit) [inline src/controllers/check.your.answers.controller.ts:5]: missing spaces around braces

File: src/controllers/check.your.answers.controller.ts [hunk 4]
```diff
@@ -11,11 +11,19 @@ export const get = async (req: Request, res: Response, next: NextFunction) => {
   if (req.session) {
     try {
       const { companyName, companyNumber } = retrieveCompanyProfileFromObjectionSession(req.session);
+      let shareIdentity: string = "No";
       const objection: Objection = await getObjection(req.session);
+      if (objection) {
+        const createdBy: CreatedBy = objection["created_by"];
+        if (createdBy) {
+          shareIdentity = (createdBy.shareIdentity === true) ? "Yes" : "No";
+        }
```
Comment (markpit) [inline src/controllers/check.your.answers.controller.ts:20]: what happens if there's no createdBy?

File: src/controllers/check.your.answers.controller.ts [hunk 5]
```diff
@@ -11,11 +11,19 @@ export const get = async (req: Request, res: Response, next: NextFunction) => {
   if (req.session) {
     try {
       const { companyName, companyNumber } = retrieveCompanyProfileFromObjectionSession(req.session);
+      let shareIdentity: string = "No";
       const objection: Objection = await getObjection(req.session);
+      if (objection) {
```
Comment (markpit) [inline src/controllers/check.your.answers.controller.ts:16]: this new code needs to be covered by unit test

File: src/controllers/check.your.answers.controller.ts (hunk 3 above)
Comment (markpit) [inline src/controllers/check.your.answers.controller.ts:5]: missing spaces around braces on import, i think eslint has reverted the tslint rule

File: src/controllers/check.your.answers.controller.ts (hunk 4 above)
Comment (markpit) [reply src/controllers/check.your.answers.controller.ts:20]: > what happens if there's no createdBy?

I can see the default "No" above, I must have missed it

File: src/controllers/check.your.answers.controller.ts (hunk 3 above)
Comment (mattch1) [reply src/controllers/check.your.answers.controller.ts]: Spaces

File: src/controllers/check.your.answers.controller.ts (hunk 5 above)
Comment (mattch1) [reply src/controllers/check.your.answers.controller.ts:16]: Updated tests to check that the fields are present.

**Review by markpit — APPROVED** [review]

**Review by ICoxCH — APPROVED** [review]

**Pull request merged**